### PR TITLE
Add v4 config to xla_dist

### PR DESCRIPTION
**Pull request by JackCaoG**
Currently xla_dist does not support v4 since mesh config is missing.

### CODE REVIEW

File: torch_xla/distributed/xla_dist.py [hunk 1]
```diff
@@ -316,6 +316,17 @@ def _tpuvm_env_vars_cmd(self, worker_idx):
         'v3-512': '8,8,1',
         'v3-1024': '8,16,1',
         'v3-2048': '16,16,1',
+        # v4
```
[the remark this answers is not in the document]
Comment (miladm) [reply torch_xla/distributed/xla_dist.py:319]: nit: suggesting to include a comment defining dimensions:
`number of host machines`, `number of chips per host machine`, `number of cores per chip`

**Review by miladm — COMMENTED** [review]
Thanks @JackCaoG.

Added a house cleaning comment. 
LGTM otherwise.

File: torch_xla/distributed/xla_dist.py [hunk 2]
```diff
@@ -327,6 +338,13 @@ def _tpuvm_env_vars_cmd(self, worker_idx):
 
   def _env_vars_cmd(self, worker_idx):
     client_worker = self._cluster.get_client_workers()[worker_idx]
+    accelerator_gen = self._cluster.get_service_workers(
+    )[0]._machine_type.split('-')[0]
+    accelerator_gen_to_tpu_num_devices = {
```
[the remark this answers is not in the document]
Comment (JackCaoG) [reply torch_xla/distributed/xla_dist.py:343]: I don't think it is worth to put every config into yaml, especially if it only 3 lines long. `TPU_NUM_DEVICES` will most likely go away after pjrt migration too.

**Review by miladm — APPROVED** [review]
Sounds good. Let's address these issues when we handle PJRT at a later time.

Thanks @JackCaoG

**Pull request merged**